modules/sfp_spur.py

```diff
@@ -0,0 +1,237 @@
+# -*- coding: utf-8 -*-
+# -------------------------------------------------------------------------------
+# Name:         sfp_spur
+# Purpose:      Spiderfoot plugin to search spur.us API for any 
+#               malicious activity by the target
+#
+# Author:      Krishnasis Mandal <krishnasis@hotmail.com>
+#
+# Created:     12/06/2020
+# Copyright:   (c) Steve Micallef
+# Licence:     GPL
+# -------------------------------------------------------------------------------
+
+from sflib import SpiderFoot, SpiderFootPlugin, SpiderFootEvent
+from netaddr import IPNetwork
+import json
+
+class sfp_spur(SpiderFootPlugin):
+    """spur.us:Investigate,Passive:Reputation Systems:apikey:Obtain information about any malicious activities involving IP addresses found"""
+
+    opts = {
+        'api_key': '',
+        'checkaffiliates': True,
+        'subnetlookup': False,
+        'netblocklookup': True,
+        'maxnetblock': 24,
+        'maxsubnet': 24
+    }
+
+    # Option descriptions. Delete any options not applicable to this module.
+    optdescs = {
+        "api_key": "spur.us API Key",
+        'checkaffiliates': "Check affiliates?",
+        'subnetlookup': "Look up all IPs on subnets which your target is a part of?",
+        'netblocklookup': "Look up all IPs on netblocks deemed to be owned by your target for possible blacklisted hosts on the same target subdomain/domain?",
+        'maxnetblock': "If looking up owned netblocks, the maximum netblock size to look up all IPs within (CIDR value, 24 = /24, 16 = /16, etc.)",
+        'maxsubnet': "If looking up subnets, the maximum subnet size to look up all the IPs within (CIDR value, 24 = /24, 16 = /16, etc.)"
+    }
+
+    results = None
+    errorState = False  
+
+    def setup(self, sfc, userOpts=dict()):
+        self.sf = sfc
+        self.results = self.tempStorage()
+
+        for opt in list(userOpts.keys()):
+            self.opts[opt] = userOpts[opt]
+
+    # What events is this module interested in for input
+    # For a list of all events, check sfdb.py.
+    def watchedEvents(self):
+        return ["IP_ADDRESS", "NETBLOCK_OWNER", "NETBLOCK_MEMBER",
+            "AFFILIATE_IPADDR"]
+
+    # What events this module produces
+    def producedEvents(self):
+        return ["IP_ADDRESS", "MALICIOUS_IPADDR", "RAW_RIR_DATA",
+            "GEO_INFO", "COMPANY_NAME", "AFFILIATE_MALICIOUS_IPADDR"]
```

Comment:
`MALICIOUS_AFFILIATE_IPADDR` - you can check `sfdb.py` in future to ensure the correct names are used.